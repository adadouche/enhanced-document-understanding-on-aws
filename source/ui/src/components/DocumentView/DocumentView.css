/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

.react-tabs__tab {
    cursor: pointer;
}

.tab-box {
    cursor: pointer;
    display: inline-block;
    width: 20%;
    padding: 10px;
    text-align: center;
    border: 1px solid #ccc;
}

.document-view-box {
    padding-top: 3%;
    padding-left: 1%;
    padding-right: 1%;
}

.document-view-header {
    padding-top: 1%;
    padding-left: 20%;
    padding-bottom: -10%;
}
