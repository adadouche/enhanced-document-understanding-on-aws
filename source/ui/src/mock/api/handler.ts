// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { rest } from 'msw';

export const MOCK_CONFIG = {
    ApiEndpoint: 'https://jsonplaceholder.typicode.com/',
    AWS_REGION: 'us-east-1',
    USER_POOL_ID: 'test_user_pool_id',
    USER_POOL_CLIENT_ID: 'testClientId',
    DATA_BUCKET_NAME: 'test-bucket'
};

export const getSignedUrlResponse = {
    downloadUrl: `${MOCK_CONFIG.ApiEndpoint}fake-url`
};

const getDocumentResponse = {
    key: `${MOCK_CONFIG.ApiEndpoint}/fake-key`
};

const getCasesResponse = {
    Count: 4,
    ScannedCount: 4,
    Items: [
        {
            CASE_ID: { S: 'fake-case-id-1' },
            CASE_NAME: { S: 'fake-case-name-1' },
            DOCUMENT_ID: { S: '0000' },
            STATUS: { S: 'success' },
            USER_ID: { S: 'fake-user-id' },
            DOC_COUNT: { N: '1' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        },
        {
            CASE_ID: { S: 'fake-case-id-2' },
            CASE_NAME: { S: 'fake-case-name-2' },
            DOCUMENT_ID: { S: '0000' },
            STATUS: { S: 'success' },
            USER_ID: { S: 'fake-user-id' },
            DOC_COUNT: { N: '0' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        },
        {
            CASE_ID: { S: 'fake-case-id-3' },
            CASE_NAME: { S: 'fake-case-name-3' },
            DOCUMENT_ID: { S: '0000' },
            STATUS: { S: 'success' },
            USER_ID: { S: 'fake-user-id' },
            DOC_COUNT: { N: '0' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        },
        {
            CASE_ID: { S: 'fake-case-id-4' },
            CASE_NAME: { S: 'fake-case-name-4' },
            DOCUMENT_ID: { S: '0000' },
            STATUS: { S: 'success' },
            USER_ID: { S: 'fake-user-id' },
            DOC_COUNT: { N: '0' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        }
    ]
};

const getDocumentByCaseIdResponse = {
    Count: 1,
    ScannedCount: 1,
    Items: [
        {
            BUCKET_NAME: { S: 'fake-bucket-name-1' },
            CASE_ID: { S: 'fake-case-id-1' },
            CASE_NAME: { S: 'fake-case-name-1' },
            DOCUMENT_ID: { S: 'fake-doc-id1' },
            DOCUMENT_TYPE: { S: 'passport' },
            S3_KEY: {
                S: 'fake-s3-key.jpg'
            },
            UPLOADED_FILE_EXTENSION: { S: 'jpg' },
            UPLOADED_FILE_NAME: { S: 'fake-file-name' },
            USER_ID: { S: 'fake-user-id' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        }
    ]
};

export const handlers = [
    rest.get('/runtimeConfig.json', (_, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(MOCK_CONFIG)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}cases`, (req, res, ctx) => {
        let response: any = getCasesResponse;
        if (req.params) {
            response = {
                ...getCasesResponse,
                LastEvaluatedKey: true
            };
        }
        return res(ctx.status(200), ctx.text(JSON.stringify(response)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}case/:caseId`, (_, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(getDocumentByCaseIdResponse)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}document/:caseId/:documentId`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(getDocumentResponse)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}document/download`, (_, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(getSignedUrlResponse)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}inferences/:caseId/:documentId`, (_, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(['textract-detectText', 'entity-medical-locations'])));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}inferences/:caseId/:documentId/:inferenceType`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify({})));
    }),
    rest.post(`${MOCK_CONFIG.ApiEndpoint}document`, (_, res, ctx) => {
        return res(
            ctx.status(200),
            ctx.text(
                JSON.stringify({
                    fields: [
                        {
                            key: 'test-entry'
                        }
                    ],
                    url: 'signed-url'
                })
            )
        );
    }),
    rest.all('/signed-url', (req, res, ctx) => {
        return res(ctx.status(403));
    })
];
