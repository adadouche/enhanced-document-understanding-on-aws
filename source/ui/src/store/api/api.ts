/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the License). You may not use this file except in compliance      *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { BaseQueryFn, createApi, FetchArgs, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react';
import { generateToken, getRuntimeConfig } from '../../utils/apiHelper';

export const dynamicBaseQuery: BaseQueryFn<string | FetchArgs, unknown, FetchBaseQueryError> = async (
    args: any,
    api: any,
    extraOptions: any
) => {
    const runtimeConfig = await getRuntimeConfig();
    const rawBaseQuery = fetchBaseQuery({
        baseUrl: runtimeConfig.ApiEndpoint,
        prepareHeaders: async (headers: any) => {
            if (process.env.NODE_ENV !== 'test') {
                const token = await generateToken();
                headers.set('Authorization', `${token}`);
            }
            return headers;
        }
    });
    return rawBaseQuery(args, api, extraOptions);
};
export const api = createApi({
    reducerPath: 'api',
    baseQuery: dynamicBaseQuery,
    tagTypes: ['Cases'],
    endpoints: () => ({
       
    }),
    refetchOnMountOrArgChange: true
});

