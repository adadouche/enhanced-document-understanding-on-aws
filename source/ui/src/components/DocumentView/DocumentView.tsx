/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import './DocumentView.css';

import { AppLayout, ContentLayout, StatusIndicatorProps, Tabs } from '@cloudscape-design/components';
import React, { useEffect } from 'react';
import {
    API_NAME,
    COMPREHEND_MEDICAL_SERVICE,
    COMPREHEND_SERVICE,
    EntityTypes,
    InferenceName,
    TEXTRACT_KEY_VALUE_PAIRS,
    TEXTRACT_RAW_TEXT,
    TEXTRACT_TABLES
} from '../../utils/constants';
import {
    getDocumentKeyValuePairs,
    getDocumentLines,
    getDocumentPageCount,
    getDocumentTables
} from '../../utils/document';
import { DocumentProcessingResponse } from '../../utils/interfaces';

import { API } from 'aws-amplify';
import { generateToken } from '../DocumentTable/DocumentTable';
import EntityDetectionTab from '../EntityDetectionTab';
import TextractTab from '../TextractTab';
import { DocumentResultsInfoPanelContent } from '../../utils/info-panel-contents';
import { DocResultsPageHeader } from '../DocumentTable/full-page-header';

type DocumentViewProps = {
    selectedDocumentId: string;
    selectedCaseId: string;
    selectedDocumentFileType: string;
    selectedCaseName: string;
    selectedDocumentName: string;
    setSelectedCaseId: React.Dispatch<React.SetStateAction<string>>;
    setSelectedDocumentId: React.Dispatch<React.SetStateAction<string>>;
    setSelectedDocumentFileType: React.Dispatch<React.SetStateAction<string>>;
    setSelectedCaseName: React.Dispatch<React.SetStateAction<string>>;
    setSelectedDocumentName: React.Dispatch<React.SetStateAction<string>>;
};

let token: string;

export default function DocumentView(props: DocumentViewProps) {
    const [documentUrl, setDocumentUrl] = React.useState<string>('');
    const [currentStatus, setCurrentStatus] = React.useState<StatusIndicatorProps.Type | undefined>();
    const [documentPageCount, setDocumentPageCount] = React.useState<number>(0);
    const [currentPageNumber, setCurrentPageNumber] = React.useState(1);
    const [documentProcessingResults, setDocumentProcessingResults] = React.useState<DocumentProcessingResponse>({
        textractDetectResponse: {
            Bucket: '',
            UploadedFileName: '',
            DocumentMetadata: { Pages: 0 },
            JobStatus: '',
            AnalyzeDocumentModelVersion: ''
        },
        comprehendGenericResponse: {
            results: []
        },
        comprehendMedicalResponse: {
            results: []
        },
        comprehendPiiResponse: {
            results: []
        }
    });
    const [selectedEntities, setSelectedEntities] = React.useState<any>({
        [EntityTypes.ENTITY_STANDARD]: [],
        [EntityTypes.PII]: [],
        [EntityTypes.MEDICAL_ENTITY]: []
    });
    const [previewRedaction, setPreviewRedaction] = React.useState('');
    const [toolsOpen, setToolsOpen] = React.useState(false);

    const switchPage = (newPageNumber: number) => {
        setCurrentPageNumber(newPageNumber);
    };

    const docData = React.useMemo(() => {
        const pairs = getDocumentKeyValuePairs(documentProcessingResults, 'KEY_VALUE_SET');
        const tables = getDocumentTables(documentProcessingResults, 'TABLE');
        const lines = getDocumentLines(documentProcessingResults, 'LINE');
        const standardEntities = documentProcessingResults.comprehendGenericResponse;
        const medicalEntities = documentProcessingResults.comprehendMedicalResponse;
        const piiEntities = documentProcessingResults.comprehendPiiResponse;
        return { pairs, lines, tables, standardEntities, medicalEntities, piiEntities };
    }, [documentProcessingResults]);

    useEffect(() => {
        props.setSelectedDocumentId(window.sessionStorage.getItem('selectedDocumentId') || '');
        props.setSelectedCaseId(window.sessionStorage.getItem('selectedCaseId') || '');
        props.setSelectedDocumentFileType(window.sessionStorage.getItem('selectedDocumentFileType') || '');
        props.setSelectedCaseName(window.sessionStorage.getItem('selectedCaseName') || '');
        props.setSelectedDocumentName(window.sessionStorage.getItem('selectedDocumentName') || '');
    }, [props]);

    const getResource = async (endpoint: string, queryStringParameters = {}) => {
        try {
            token = await generateToken();
            const response = await API.get(API_NAME, endpoint, {
                headers: {
                    Authorization: token
                },
                queryStringParameters: queryStringParameters
            });
            return response;
        } catch (err) {
            console.error(err);
        }
    };

    const formatTextract = (response: any) => {
        if (response) {
            for (let pageNumber = 0; pageNumber < response.length; pageNumber++) {
                response[pageNumber].DocumentMetadata.Pages = response.length;
                for (const block of response[pageNumber].Blocks) {
                    block.Page = pageNumber + 1;
                }
            }
        }
        return response;
    };

    const retrieveSignedUrl = async () => {
        const documentResponse = await getResource(`document/${props.selectedCaseId}/${props.selectedDocumentId}`, {
            redacted: false
        });

        const signedUrlObject = await getResource(`document/download`, { key: documentResponse.key });
        setDocumentUrl(signedUrlObject.downloadUrl);
    };

    useEffect(() => {
        const getDocumentData = async () => {
            setCurrentStatus('loading');
            let documentResponse,
                textractDetectResponse,
                textractAnalyzeResponse,
                entityGenericResponse,
                entityMedicalResponse,
                entityPiiResponse;
            let validInferences = [];
            textractDetectResponse =
                textractAnalyzeResponse =
                entityGenericResponse =
                entityMedicalResponse =
                entityPiiResponse =
                    undefined;

            try {
                if (props.selectedCaseId === '' || props.selectedDocumentId === '') {
                    setCurrentStatus('error');
                    return;
                }

                validInferences = await getResource(`inferences/${props.selectedCaseId}/${props.selectedDocumentId}`);

                documentResponse = await getResource(`document/${props.selectedCaseId}/${props.selectedDocumentId}`, {
                    redacted: false
                });

                const signedUrlObject = await getResource(`document/download`, { key: documentResponse.key });
                setDocumentUrl(signedUrlObject.downloadUrl);

                if (validInferences.includes(InferenceName.TEXTRACT_DETECT_TEXT)) {
                    textractDetectResponse = await getResource(
                        `inferences/${props.selectedCaseId}/${props.selectedDocumentId}/${InferenceName.TEXTRACT_DETECT_TEXT}`
                    );
                }

                if (validInferences.includes(InferenceName.TEXTRACT_ANALYZE_TEXT)) {
                    textractAnalyzeResponse = await getResource(
                        `inferences/${props.selectedCaseId}/${props.selectedDocumentId}/${InferenceName.TEXTRACT_ANALYZE_TEXT}`
                    );
                }

                if (validInferences.includes(InferenceName.COMPREHEND_GENERIC)) {
                    entityGenericResponse = await getResource(
                        `inferences/${props.selectedCaseId}/${props.selectedDocumentId}/${InferenceName.COMPREHEND_GENERIC}`
                    );
                }

                if (validInferences.includes(InferenceName.COMPREHEND_PII)) {
                    entityPiiResponse = await getResource(
                        `inferences/${props.selectedCaseId}/${props.selectedDocumentId}/${InferenceName.COMPREHEND_PII}`
                    );
                }

                if (validInferences.includes(InferenceName.COMPREHEND_MEDICAL)) {
                    entityMedicalResponse = await getResource(
                        `inferences/${props.selectedCaseId}/${props.selectedDocumentId}/${InferenceName.COMPREHEND_MEDICAL}`
                    );
                }

                setDocumentProcessingResults({
                    textractDetectResponse: formatTextract(textractDetectResponse),
                    textractAnalyzeResponse: formatTextract(textractAnalyzeResponse),
                    comprehendGenericResponse: entityGenericResponse,
                    comprehendMedicalResponse: entityMedicalResponse,
                    comprehendPiiResponse: entityPiiResponse
                });
                setCurrentStatus('success');
            } catch (error) {
                console.error(
                    `Error in retrieving document and inferences with ${[
                        props.selectedCaseId,
                        props.selectedDocumentId
                    ]}: ${error}`
                );
                setCurrentStatus('error');
            }
        };

        getDocumentData();
    }, [props.selectedCaseId, props.selectedDocumentId]);

    useEffect(() => {
        setDocumentPageCount(getDocumentPageCount(documentProcessingResults, 'LINE'));
    }, [documentProcessingResults]);

    const mainTabs = [
        {
            label: 'Entity Detection',
            id: 'entityDetection',
            content: (
                <EntityDetectionTab
                    selectedDocumentFileType={props.selectedDocumentFileType}
                    selectedDocumentUrl={documentUrl}
                    standardEntities={docData.standardEntities}
                    medicalEntities={docData.medicalEntities}
                    piiEntities={docData.piiEntities}
                    documentPageCount={documentPageCount}
                    currentPageNumber={currentPageNumber}
                    switchPage={switchPage}
                    comprehendService={COMPREHEND_SERVICE}
                    entityType={EntityTypes.ENTITY_STANDARD}
                    selectedDocumentId={props.selectedDocumentId}
                    selectedCaseId={props.selectedCaseId}
                    currentStatus={currentStatus}
                    selectedEntities={selectedEntities}
                    setSelectedEntities={setSelectedEntities}
                    previewRedaction={previewRedaction}
                    setPreviewRedaction={setPreviewRedaction}
                    retrieveSignedUrl={retrieveSignedUrl}
                    dataTestId="entity-detection-tab"
                />
            )
        },
        {
            label: 'Medical Entity Detection',
            id: 'medicalEntityDetection',
            content: (
                <EntityDetectionTab
                    selectedDocumentFileType={props.selectedDocumentFileType}
                    selectedDocumentUrl={documentUrl}
                    standardEntities={docData.standardEntities}
                    medicalEntities={docData.medicalEntities}
                    piiEntities={docData.piiEntities}
                    documentPageCount={documentPageCount}
                    currentPageNumber={currentPageNumber}
                    switchPage={switchPage}
                    comprehendService={COMPREHEND_MEDICAL_SERVICE}
                    entityType={EntityTypes.MEDICAL_ENTITY}
                    selectedDocumentId={props.selectedDocumentId}
                    selectedCaseId={props.selectedCaseId}
                    currentStatus={currentStatus}
                    selectedEntities={selectedEntities}
                    setSelectedEntities={setSelectedEntities}
                    previewRedaction={previewRedaction}
                    setPreviewRedaction={setPreviewRedaction}
                    retrieveSignedUrl={retrieveSignedUrl}
                    dataTestId="medical-entity-detection-tab"
                />
            )
        },
        {
            label: 'PII Detection',
            id: 'piiDetection',
            content: (
                <EntityDetectionTab
                    selectedDocumentFileType={props.selectedDocumentFileType}
                    selectedDocumentUrl={documentUrl}
                    standardEntities={docData.standardEntities}
                    medicalEntities={docData.medicalEntities}
                    piiEntities={docData.piiEntities}
                    documentPageCount={documentPageCount}
                    currentPageNumber={currentPageNumber}
                    switchPage={switchPage}
                    comprehendService={COMPREHEND_SERVICE}
                    entityType={EntityTypes.PII}
                    selectedDocumentId={props.selectedDocumentId}
                    selectedCaseId={props.selectedCaseId}
                    currentStatus={currentStatus}
                    selectedEntities={selectedEntities}
                    setSelectedEntities={setSelectedEntities}
                    previewRedaction={previewRedaction}
                    setPreviewRedaction={setPreviewRedaction}
                    retrieveSignedUrl={retrieveSignedUrl}
                    dataTestId="pii-detection-tab"
                />
            )
        },
        {
            label: 'Raw Text',
            id: 'textractRawText',
            content: (
                <TextractTab
                    selectedDocumentFileType={props.selectedDocumentFileType}
                    selectedDocumentUrl={documentUrl}
                    documentLines={docData.lines}
                    kvPairs={docData.pairs}
                    tables={docData.tables}
                    documentPageCount={documentPageCount}
                    currentPageNumber={currentPageNumber}
                    switchPage={switchPage}
                    textractOutputType={TEXTRACT_RAW_TEXT}
                    currentStatus={currentStatus}
                    retrieveSignedUrl={retrieveSignedUrl}
                    dataTestId="textract-raw-text-tab"
                />
            )
        },
        {
            label: 'Key-Value Pairs',
            id: 'textractKeyValuePairs',
            content: (
                <TextractTab
                    selectedDocumentFileType={props.selectedDocumentFileType}
                    selectedDocumentUrl={documentUrl}
                    documentLines={docData.lines}
                    kvPairs={docData.pairs}
                    tables={docData.tables}
                    documentPageCount={documentPageCount}
                    currentPageNumber={currentPageNumber}
                    switchPage={switchPage}
                    textractOutputType={TEXTRACT_KEY_VALUE_PAIRS}
                    currentStatus={currentStatus}
                    retrieveSignedUrl={retrieveSignedUrl}
                    dataTestId="textract-key-value-pairs-tab"
                />
            )
        },
        {
            label: 'Tables',
            id: 'textractTables',
            content: (
                <TextractTab
                    selectedDocumentFileType={props.selectedDocumentFileType}
                    selectedDocumentUrl={documentUrl}
                    documentLines={docData.lines}
                    kvPairs={docData.pairs}
                    tables={docData.tables}
                    documentPageCount={documentPageCount}
                    currentPageNumber={currentPageNumber}
                    switchPage={switchPage}
                    textractOutputType={TEXTRACT_TABLES}
                    currentStatus={currentStatus}
                    retrieveSignedUrl={retrieveSignedUrl}
                    dataTestId="textract-tables-tab"
                />
            )
        }
    ];

    return (
        <AppLayout
            contentType="dashboard"
            navigationHide
            onToolsChange={({ detail }) => {
                setToolsOpen(detail.open);
            }}
            toolsOpen={toolsOpen}
            tools={<DocumentResultsInfoPanelContent />}
            content={
                <ContentLayout
                    header={
                        <div className="document-view-header">
                            <DocResultsPageHeader
                                onInfoLinkClick={() => {
                                    setToolsOpen(true);
                                }}
                                breadCrumbItems={{
                                    selectedCaseName: props.selectedCaseName,
                                    selectedDocumentName: props.selectedDocumentName
                                }}
                            />
                        </div>
                    }
                >
                    <div data-testid="document-view-box" className="document-view-box">
                        <Tabs tabs={mainTabs} data-testid="document-view-tabs" />
                    </div>
                </ContentLayout>
            }
            data-testid="document-view-app-layout"
        />
    );
}
