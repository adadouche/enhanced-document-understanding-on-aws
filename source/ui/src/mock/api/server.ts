// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { handlers } from "./handler";
import { setupServer } from "msw/node";
export const server = setupServer(...handlers);