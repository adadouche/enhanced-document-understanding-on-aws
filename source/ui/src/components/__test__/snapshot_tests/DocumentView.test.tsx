/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { API } from '@aws-amplify/api';
import DocumentView from '../../DocumentView/DocumentView';

const mockAPI = {
    get: jest.fn(),
    post: jest.fn()
};
jest.mock('@aws-amplify/api');
API.get = mockAPI.get;
API.post = mockAPI.post;
const documentViewProps = {
    selectedDocumentId: 'fake-doc-id',
    selectedCaseId: 'fake-case-id',
    selectedDocumentFileType: 'jpg',
    selectedCaseName: 'fake-case-name',
    selectedDocumentName: 'fake-doc-name',
    setSelectedCaseId: jest.fn,
    setSelectedDocumentId: jest.fn,
    setSelectedDocumentFileType: jest.fn,
    setSelectedCaseName: jest.fn,
    setSelectedDocumentName: jest.fn
};

beforeEach(() => {
    mockAPI.get.mockReset();
});

const getDocumentResponse = {
    key: 'fake-key'
};

const getSignedUrlResponse = {
    downloadUrl: 'fake-url'
};

jest.mock('@cloudscape-design/components', () => {
    const Components = jest.genMockFromModule('@cloudscape-design/components') as any;
    for (const componentName of Object.keys(Components)) {
        Components[componentName] = componentName;
    }
    return Components;
});

test('Snapshot test', async () => {
    mockAPI.get.mockResolvedValueOnce(getDocumentResponse).mockResolvedValueOnce(getSignedUrlResponse);
    const tree = renderer.create(<DocumentView {...documentViewProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});
