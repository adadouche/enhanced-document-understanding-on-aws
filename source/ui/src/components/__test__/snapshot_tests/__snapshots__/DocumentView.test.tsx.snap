// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<AppLayout
  content={
    <ContentLayout
      header={
        <div
          className="document-view-header"
        >
          <DocResultsPageHeader
            breadCrumbItems={
              Object {
                "selectedCaseName": "fake-case-name",
                "selectedDocumentName": "fake-doc-name",
              }
            }
            onInfoLinkClick={[Function]}
          />
        </div>
      }
    >
      <div
        className="document-view-box"
        data-testid="document-view-box"
      >
        <Tabs
          data-testid="document-view-tabs"
          tabs={
            Array [
              Object {
                "content": <EntityDetectionTab
                  comprehendService="COMPREHEND"
                  currentPageNumber={1}
                  dataTestId="entity-detection-tab"
                  documentPageCount={0}
                  entityType="entity-standard"
                  medicalEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  piiEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  previewRedaction=""
                  retrieveSignedUrl={[Function]}
                  selectedCaseId="fake-case-id"
                  selectedDocumentFileType="jpg"
                  selectedDocumentId="fake-doc-id"
                  selectedDocumentUrl=""
                  selectedEntities={
                    Object {
                      "entity-medical": Array [],
                      "entity-pii": Array [],
                      "entity-standard": Array [],
                    }
                  }
                  setPreviewRedaction={[Function]}
                  setSelectedEntities={[Function]}
                  standardEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  switchPage={[Function]}
                />,
                "id": "entityDetection",
                "label": "Entity Detection",
              },
              Object {
                "content": <EntityDetectionTab
                  comprehendService="COMPREHEND_MEDICAL"
                  currentPageNumber={1}
                  dataTestId="medical-entity-detection-tab"
                  documentPageCount={0}
                  entityType="entity-medical"
                  medicalEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  piiEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  previewRedaction=""
                  retrieveSignedUrl={[Function]}
                  selectedCaseId="fake-case-id"
                  selectedDocumentFileType="jpg"
                  selectedDocumentId="fake-doc-id"
                  selectedDocumentUrl=""
                  selectedEntities={
                    Object {
                      "entity-medical": Array [],
                      "entity-pii": Array [],
                      "entity-standard": Array [],
                    }
                  }
                  setPreviewRedaction={[Function]}
                  setSelectedEntities={[Function]}
                  standardEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  switchPage={[Function]}
                />,
                "id": "medicalEntityDetection",
                "label": "Medical Entity Detection",
              },
              Object {
                "content": <EntityDetectionTab
                  comprehendService="COMPREHEND"
                  currentPageNumber={1}
                  dataTestId="pii-detection-tab"
                  documentPageCount={0}
                  entityType="entity-pii"
                  medicalEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  piiEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  previewRedaction=""
                  retrieveSignedUrl={[Function]}
                  selectedCaseId="fake-case-id"
                  selectedDocumentFileType="jpg"
                  selectedDocumentId="fake-doc-id"
                  selectedDocumentUrl=""
                  selectedEntities={
                    Object {
                      "entity-medical": Array [],
                      "entity-pii": Array [],
                      "entity-standard": Array [],
                    }
                  }
                  setPreviewRedaction={[Function]}
                  setSelectedEntities={[Function]}
                  standardEntities={
                    Object {
                      "results": Array [],
                    }
                  }
                  switchPage={[Function]}
                />,
                "id": "piiDetection",
                "label": "PII Detection",
              },
              Object {
                "content": <TextractTab
                  currentPageNumber={1}
                  dataTestId="textract-raw-text-tab"
                  documentLines={Array []}
                  documentPageCount={0}
                  kvPairs={Array []}
                  retrieveSignedUrl={[Function]}
                  selectedDocumentFileType="jpg"
                  selectedDocumentUrl=""
                  switchPage={[Function]}
                  tables={Array []}
                  textractOutputType="TEXTRACT_RAW_TEXT"
                />,
                "id": "textractRawText",
                "label": "Raw Text",
              },
              Object {
                "content": <TextractTab
                  currentPageNumber={1}
                  dataTestId="textract-key-value-pairs-tab"
                  documentLines={Array []}
                  documentPageCount={0}
                  kvPairs={Array []}
                  retrieveSignedUrl={[Function]}
                  selectedDocumentFileType="jpg"
                  selectedDocumentUrl=""
                  switchPage={[Function]}
                  tables={Array []}
                  textractOutputType="TEXTRACT_KEY_VALUE_PAIRS"
                />,
                "id": "textractKeyValuePairs",
                "label": "Key-Value Pairs",
              },
              Object {
                "content": <TextractTab
                  currentPageNumber={1}
                  dataTestId="textract-tables-tab"
                  documentLines={Array []}
                  documentPageCount={0}
                  kvPairs={Array []}
                  retrieveSignedUrl={[Function]}
                  selectedDocumentFileType="jpg"
                  selectedDocumentUrl=""
                  switchPage={[Function]}
                  tables={Array []}
                  textractOutputType="TEXTRACT_TABLES"
                />,
                "id": "textractTables",
                "label": "Tables",
              },
            ]
          }
        />
      </div>
    </ContentLayout>
  }
  contentType="dashboard"
  data-testid="document-view-app-layout"
  navigationHide={true}
  onToolsChange={[Function]}
  tools={<DocumentResultsInfoPanelContent />}
  toolsOpen={false}
/>
`;
